### DISCUSSION TITLE
Storage bucked INSERT filtering with mimetype

### DISCUSSION THREAD

**Discussion by TheBlueOompaLoompa** [supabase/supabase#1538] [Questions]
I have a storage bucket that I want to only take files with the mime type that contains _image_.
Here is my attempt: `((metadata ->> 'mimetype'::text) ~ 'image'::text)`

**Comment by inian** [comment]
Hi @TheBlueOompaLoompa, are you looking for a way to have an allowlist of mimetypes for the entire bucket? 

Currently adding policies based on metadata for INSERTs is not supported. This is because we find out the metadata of the object (like size and mimetype) only _after_ the object is uploaded  since we directly stream the request to s3. This is why policies for INSERT based on size is [not supported ](https://github.com/supabase/supabase/discussions/1123)as well. 
The rough flow for upload is like this

- check if user has permission to insert a row into the objects table. if the user is able to insert, we assume they have the right permissions
- upload the object and get metadata after it is uploaded from s3
- update table with the metadata 

For now you might have to add the check on the client side or check the mimetype after the upload operation has finished.
